processor/src/main/java/org/mapstruct/ap/internal/processor/DefaultVersionInformation.java

```diff
@@ -37,15 +38,19 @@
     private final String runtimeVersion;
     private final String runtimeVendor;
     private final String compiler;
+    private final boolean sourceVersionAtLeast9;
     private final boolean eclipseJDT;
     private final boolean javac;
 
-    DefaultVersionInformation(String runtimeVersion, String runtimeVendor, String compiler) {
+    DefaultVersionInformation(String runtimeVersion, String runtimeVendor, String compiler,
+        SourceVersion sourceVersion) {
         this.runtimeVersion = runtimeVersion;
         this.runtimeVendor = runtimeVendor;
         this.compiler = compiler;
         this.eclipseJDT = compiler.startsWith( COMPILER_NAME_ECLIPSE_JDT );
         this.javac = compiler.startsWith( COMPILER_NAME_JAVAC );
+        // If the difference between the source version and RELEASE_6 is more that 2 than we are at least on 9
+        this.sourceVersionAtLeast9 = sourceVersion.compareTo( SourceVersion.RELEASE_6 ) > 2;
```

Comment:
Yes from the `Enum` javadoc:

>The natural order implemented by this method (`compareTo`) is the order in which the constants are declared.

On top of that the implementation is:

```
public final int compareTo(E o) {
    Enum<?> other = (Enum<?>)o;
    Enum<E> self = this;
    if (self.getClass() != other.getClass() && // optimization
        self.getDeclaringClass() != other.getDeclaringClass())
        throw new ClassCastException();
    return self.ordinal - other.ordinal;
}
```

If Java changes the `Enum` compareTo then it would be a problem, but I doubt they'll do that. Google AutoValue uses the same logic, and I presume other processors as well